daemon/src/main/java/org/apache/maven/cli/DaemonMavenParser.java

```diff
@@ -0,0 +1,105 @@
+
+public class DaemonMavenParser extends BaseMavenParser<MavenOptions, MavenInvokerRequest<MavenOptions>> {
+    @Override
+    protected DefaultMavenInvokerRequest<MavenOptions> getInvokerRequest(LocalContext context) {
+        return new DefaultMavenInvokerRequest<>(
+                context.parserRequest,
+                context.cwd,
+                context.installationDirectory,
+                context.userHomeDirectory,
+                context.userProperties,
+                context.systemProperties,
+                context.topDirectory,
+                context.rootDirectory,
+                context.parserRequest.in(),
+                context.parserRequest.out(),
+                context.parserRequest.err(),
+                context.extensions,
+                (DaemonMavenOptions) context.options);
+    }
+
+    @Override
+    protected Path getRootDirectory(LocalContext context) throws ParserException {
+        Map<String, String> env = context.parserRequest
+                .lookup()
+                .lookup(DaemonMavenCling.Environment.class)
+                .getEnvironment();
+        System.setProperty("maven.multiModuleProjectDirectory", env.get("maven.multiModuleProjectDirectory"));
+        return super.getRootDirectory(context);
+    }
+
+    @Override
+    protected MavenOptions parseArgs(String source, List<String> args) throws ParserException {
+        try {
+            return CommonsCliDaemonMavenOptions.parse(source, args.toArray(new String[0]));
+        } catch (ParseException e) {
+            throw new ParserException("Failed to parse source " + source, e.getCause());
+        }
+    }
+
+    @Override
+    protected MavenOptions assembleOptions(List<MavenOptions> parsedOptions) {
+        return LayeredDaemonMavenOptions.layerDaemonMavenOptions(
+                parsedOptions.stream().map(o -> (DaemonMavenOptions) o).toList());
+    }
+
+    @Override
+    protected Map<String, String> populateSystemProperties(LocalContext context) throws ParserException {
+        HashMap<String, String> systemProperties = new HashMap<>(super.populateSystemProperties(context));
+        Map<String, String> env = context.parserRequest
+                .lookup()
+                .lookup(DaemonMavenCling.Environment.class)
+                .getEnvironment();
+        systemProperties.putAll(env);
+        return systemProperties;
+    }
+
+    @Override
+    protected List<CoreExtension> readCoreExtensionsDescriptor(LocalContext context) {
+        return Stream.of(Environment.MVND_CORE_EXTENSIONS.asString().split(";"))
+                .filter(s -> s != null && !s.isEmpty())
+                .map(s -> {
+                    String[] parts = s.split(":");
+                    return CoreExtension.newBuilder()
+                            .groupId(parts[0])
+                            .artifactId(parts[1])
+                            .version(parts[2])
+                            .build();
```

Comment:
This needs to be extended, as extensions.xml may have more options, like classloading?

Reply:
This is for another PR, as this PR did not alter anything in this area